components/suggestion/command_provider.jsx

```diff
@@ -130,14 +131,20 @@ export default class CommandProvider extends Provider {
 
     handleWebapp(pretext, resultCallback) {
         const command = pretext.toLowerCase();
-        const teamId = getCurrentTeamId(store.getState());
+
+        const selectedPost = getSelectedPost(store.getState());
+        let rootId;
+        if (this.rootId) {
+            rootId = selectedPost.root_id ? selectedPost.root_id : selectedPost.id;
+        }
```

Comment:
Do we need the `this.rootId` at all? When you are not in the RHS does `getSelectedPost` return `null`? Wondering if we can just use that as to determine if it's in the RHS

Reply:
I think we do -- when you are in a different center channel, but the RHS is open, `getSelectedPost` will return the post selected in the RHS.

Reply:
Right, how about when it's closed?

Reply:
Then it returns the `type: "system_fake_parent_deleted"` post.

Reply:
How about `getIsRhsOpen`?

Reply:
But we still wouldn't know if we're typing in the center channel textbox or the RHS checkbox.

Reply:
Sorry, I guess I'm not following how `rootId` guarantees they have not switched between the RHS and center channel textbox if the `CommandProvider` is not recreated

Reply:
It's not the same CommandProvider -- one is created for each textbox.

Do you think I should change that root_id to something like `isInRHS`? I just didn't like introducing booleans for information we already have. But I can see why that might be better.

Reply:
Yes, I think using the `this.rootId` as a flag and then using another value for it when you pass it as an args is confusing. Renaming that would help clear this a bit

Reply:
Great, done!